### DISCUSSION TITLE
Support page dimension setting to allow portrait and landscape printing without visual rotation

### DISCUSSION THREAD

**Discussion by alonbl** [pdfarranger/pdfarranger#844] [General]
Hello,

Maybe I missed this functionality.

Is there any way to specify page attributes so when printing the document the layout will be applied?

For example if I have 3 pages:
1. Portrait
2. Landscape
3. Portrait

When viewing in a computer, I would like to see these in human native position (not rotated).
When printed, the 2nd page should be rotated based on page attribute.

I believe that it should be possible if a page attribute is defined per each page.

Is this possible using pdf arranger?

Thanks,
Alon

**Comment by JunglePredator** [comment]
PDF Arranger can change the pdf's page orientation with 'rotate left' or 'rotate right' in the context menu.

I don't believe there's any mechanism in the pdf format that does what you're asking for; I'm not a pdf structure expert by any means however & someone else might be able to discern an option.

However... depending on how you're printing you can possibly get this behaviour; if the pdf is set to a human readable landscape format some printers or print devices will squeeze  the page into a portrait sized page instead of printing a rotated page. Check the documentation of your printer.

You can check https://developer.apple.com/documentation/pdfkit/pdfpage for information on pdf structure to see if there's a way to do it  via pdf instructions.

For messing with pdfs, since I can't afford a fancy pdf editor, I usually use cpdf @ http://www.coherentpdf.com/

**Comment by alonbl** [comment]
I think that the landscape/portrait is a property of the page, rotating pages is a "workaround", see [1] as an example. Maybe this feature of postscript is not used any more...

Please see the attached word document[2] and its conversion to pdf[3], the landscape page is presented as not rotated while the properties of the page mark it as landscape and will be rotated when printed.

[1] https://pages.cs.wisc.edu/~rjl/landscape/
[2] [a.docx](https://github.com/pdfarranger/pdfarranger/files/12541083/a.docx)
[3] [a.pdf](https://github.com/pdfarranger/pdfarranger/files/12541082/a.pdf)

**Comment by jeromerobert** [comment]
Sorry but I cannot find any landscape / portrait property in your PDF file nor in the PDF specification. I think JunglePredator is right. 

Indeed the second page is not rotated it just have the `/MediaBox [ 0 0 595.32 841.92 ]` attribute while the 2 others have `[ 0 0 841.92 595.32 ]`.

**Comment by alonbl** [comment]
Thanks for looking into it.

May I rephrase this PR?

Allow to set the page demension via the `/MediaBox` attribute, so that when viewing the PDF the page orietation will be human but when printed it will be rotated per the page dimension.

The result will be the same...

**Comment by jeromerobert** [comment]
No. It's already possible to set the mediabox. The reason why the document get rotated per the page dimension when printing is because the printer is smart enough to do it.
